docs/_threats/tr-7.md

```diff
@@ -1,14 +1,65 @@
+  - FFIEC Appendix J:
+    - "Strengthening the Resilience of Outsourced Technology Services"
 ---
 
-Hosted LLMs typical have usage limits, but often lack the sophisticated controls needed to manage this.
+RAG systems are proliferating due to the low barrier of entry compared to traditional chatbot technologies. RAG has applicability for internal users as well as supporting customer-related use cases.
 
-Given the commercial model of LLM services is typically on a per usage basis, there is the risk that significant resources can be expended via Chat App if uncontrolled access is allowed. e.g. DOS, injecting very large contexts, brute forcing to elicit very long responses, etc.
+Many LLMs require GPU compute infrastructure to provide an acceptable level of responsiveness. Furthermore, some of the best-performing LLMs are proprietary. Therefore, the path of least resistance is to call out to a Technology Service Provider (TSP).
+
+However, applications that index sensitive information may use on-prem GPUs because:
+
+1. The risks associated with data leakage are too high.
+2. The administrative barriers for getting approval to use a cloud provider are too high.
+
+Using on-prem GPU puts the onus on the system owner to ensure that VRAM exhaustion does not occur. Current GPUs use soldered VRAM to achieve maximum memory bandwidth and are not expandable. Most LLM libraries will fail when a memory allocation fails.
+
+# Key Risks
+
+-	**Denial of Wallet (DoW)**: Over-usage that results in slowdowns, unexpected costs, or outages. Examples include:
+    - Unexpectedly long prompts due to large chunk sizes or many chunks packed into the prompt.
+        - This could be exacerbated by systems that work with multimedia content.
+        - LLM attacks to exfiltrate training data may also generate the maximum number of tokens.
+    - Scripts which repeatedly hit API endpoints without built-in throttling.
+        - This may result from Strats searching for new sources of "signal".
+    - Agentic systems which may generate additional LLM / search calls that were not taken into account in the initial design / capacity planning. 
+-	**TSP Outages**:
+    - Some LLM providers might not have sufficient technical maturity to provide uptime to meet target SLA.
+    - Inability to fail over to an equivalent LLM back end due to proprietary model / lack of capacity at other TSPs. (FFIEC Appendix J).
+- **On-Prem Capacity**:
+    - Long lead time to add GPUs to data centers may result in prolonged performance degredation.
+    - BCP site may not have equivalent GPU capacity.
+    - Load testing new LLM serving libraries and settings.
+- **VRAM Exhaustion**:
+    - VRAM utilization can increase for a variety of reasons:
+        - new release of LLM library has a memory leak
+        - new caching techniques to improve throughput at the cost of VRAM
+        - setting a longer context length
+        - etc.
+
+# Recommendations:
```

Comment:
threats should describe the problem, with mitigations described via the associated controls. Can you consider whether your recommendations are covered by our existing controls, or whether additional ones are needed?

Reply:
I'll take out that section.  I'll look into whether the first couple of points can go into CT-8.